tests/test_machine_only_rules.py

```diff
@@ -59,10 +61,13 @@ def check_ds(ds_path, what, input_elems):
             idref = p.get("idref")
             if idref == machine_cpe:
                 machine_platform = True
-        if not machine_platform:
-            sys.stderr.write("%s %s in %s is missing <platform> element" %
+        # If the rule already has any platform,
+        # it is not required to inherit machine CPE from its parent group
+        if not platforms and not machine_platform:
+            sys.stderr.write("\n%s %s in %s is missing <platform> element" %
                              (what, elem_short_id, ds_path))
             return False
```

Comment:
In this way, we will get only the first rule which fails the test. I think it would be nice if we could get all the rules. Would it be difficult to implement it?